src/crimson/net/SocketMessenger.cc

```diff
@@ -175,6 +175,10 @@ seastar::future<> SocketMessenger::shutdown()
     return seastar::parallel_for_each(connections, [] (auto conn) {
       return conn.second->close_clean(false);
     });
+  }).then([this] {
+    return seastar::parallel_for_each(closing_conns, [] (auto conn) {
+      return conn->closing_future();
```

Comment:
`return conn->close_clean(false);` should also work here. Seems to me there is no need to introduce a new interface `closing_future()`, as well as changing the type of `Protocol::close_ready`.